web/src/client/software.js

```diff
@@ -348,6 +246,73 @@ class ProductBaseClient {
       }
     });
   }
+
+  /**
+   * Returns the registration of the selected product.
+   *
+   * @return {Promise<Registration>}
+   */
+  async getRegistration() {
+    const response = await this.client.get("/software/registration");
+    if (!response.ok) {
+      console.log("Failed to get registration config: ", response);
+      return { requirement: "unknown", code: null, email: null };
+    }
+    const config = await response.json();
+
+    const requirement = config.requirement;
+    const code = config.key;
+    const email = config.email;
+
+    const registration = { requirement, code, email };
+    if (code.length === 0) registration.code = null;
+    if (email.length === 0) registration.email = null;
+
+    return registration;
+  }
+
+  /**
+   * Tries to register the selected product.
+   *
+   * @param {string} code
+   * @param {string} [email]
+   * @returns {Promise<ActionResult>}
+   */
+  async register(code, email = "") {
+    const response = await this.client.post("/software/registration", { key: code, email });
+
+    return {
+      success: response.ok,
+      message: "", // TODO: pass message and code in body
+    };
+  }
+
+  /**
+   * Tries to deregister the selected product.
+   *
+   * @returns {Promise<ActionResult>}
+   */
+  async deregister() {
+    const response = await this.client.delete("/software/registration");
+
+    return {
+      success: response.ok,
+      message: "", // TODO: how to get message?
```

Comment:
What does it mean?

Reply:
well, it calls delete. And if it failed we need to somehow get error code. I am not sure if way would be using body in Delete and 200 code?

Reply:
Sure, we need the error code. My question was more about why is this a TODO? If we cannot do it right now, then it is fine.

About the answer, I would use "422 Unprocessable Content" (perhaps there is a better code) and put the error code in the body.